src/test/scala/com/databricks/labs/mosaic/expressions/geometry/ST_IntersectsBehaviors.scala

```diff
@@ -0,0 +1,123 @@
+                mosaic_explode(col("wkt"), resolution).alias("right_index"),
+                col("wkt").alias("right_wkt")
+            )
+
+        val result = left
+            .join(
+                right,
+                col("left_index.index_id") === col("right_index.index_id")
+            )
+            .groupBy(
+                "left_id",
+                "right_id"
+            )
+            .agg(
+                st_intersects_aggregate(col("left_index"), col("right_index")).alias("agg_intersects"),
+                first("left_wkt").alias("left_wkt"),
+                first("right_wkt").alias("right_wkt")
+            )
+            .withColumn(
+                "flat_intersects",
+                st_intersects(col("left_wkt"), col("right_wkt"))
+            )
+            .withColumn(
+                "comparison",
+                col("flat_intersects") === col("agg_intersects")
+            )
+
+        result.select("comparison").collect().map(_.getBoolean(0)).forall(identity) shouldBe true
+
+        left.createOrReplaceTempView("left")
+        right.createOrReplaceTempView("right")
+
+        val result2 = spark.sql("""
+                                  |SELECT ST_INTERSECTS_AGGREGATE(LEFT_INDEX, RIGHT_INDEX)
+                                  |FROM LEFT
+                                  |INNER JOIN RIGHT ON LEFT_INDEX.INDEX_ID == RIGHT_INDEX.INDEX_ID
+                                  |GROUP BY LEFT_ID, RIGHT_ID
+                                  |""".stripMargin)
+
+        result2.collect().length should be > 0
+    }
+
+    def intersectsCodegen(indexSystem: IndexSystem, geometryAPI: GeometryAPI): Unit = {
+        spark.sparkContext.setLogLevel("FATAL")
+        val mc = MosaicContext.build(indexSystem, geometryAPI)
+        val sc = spark
+        import mc.functions._
+        import sc.implicits._
+        mc.register(spark)
+
+        val result = mocks
+            .getWKTRowsDf(mc)
+            .select(st_intersects($"wkt", $"wkt"))
+            .as[Boolean]
+
+        val queryExecution = result.queryExecution
+        val plan = queryExecution.executedPlan
+
+        val wholeStageCodegenExec = plan.find(_.isInstanceOf[WholeStageCodegenExec])
+
+        wholeStageCodegenExec.isDefined shouldBe true
+
+        val codeGenStage = wholeStageCodegenExec.get.asInstanceOf[WholeStageCodegenExec]
+        val (_, code) = codeGenStage.doCodeGen()
+
+        noException should be thrownBy CodeGenerator.compile(code)
+    }
+
+    def auxiliaryMethods(indexSystem: IndexSystem, geometryAPI: GeometryAPI): Unit = {
+        spark.sparkContext.setLogLevel("FATAL")
+        val mc = MosaicContext.build(indexSystem, geometryAPI)
+        mc.register(spark)
+
+        val stIntersects =
+            ST_Intersects(lit("POLYGON (1 1, 2 2, 3 3, 4 4, 1 1)").expr, lit("POLYGON (1 2, 2 2, 3 3, 4 2, 1 2)").expr, geometryAPI.name)
+
+        stIntersects.left shouldEqual lit("POLYGON (1 1, 2 2, 3 3, 4 4, 1 1)").expr
```

Comment:
it isn't, expr.left returns the left child expression.
this is an auxiliary test to avoid coverage degradation.
IMHO it is a dummy test but it keeps green lights on.